lib/ansible/galaxy/collection.py

```diff
@@ -1384,7 +1384,19 @@ def _download_file(url, b_path, expected_hash, validate_certs, headers=None):
 
 def _extract_tar_dir(tar, dirname, b_dest):
     """ Extracts a directory from a collection tar. """
-    tar_member = tar.getmember(to_native(dirname, errors='surrogate_or_strict'))
+    try:
+        tar_member = tar.getmember(to_native(dirname, errors='surrogate_or_strict'))
+    except KeyError:
+        # Some archives contain directories that end with '/'
+        if not dirname.endswith(os.path.sep):
+            dirname += os.path.sep
+
+        try:
+            tar_member = tar.getmember(to_native(dirname, errors='surrogate_or_strict'))
+        except KeyError:
+            # If we still can't find the member to extract, raise a nice error.
+            raise AnsibleError("Unable to extract '%s' from collection" % to_native(dirname, errors='surrogate_or_strict'))
+
```

Comment:
I don't want to hold up the PR but you could use a loop instead of a nested KeyError like

```suggestion
    member_names = [to_native(dirname, errors='surrogate_or_strict')]
    if not member_names[-1].endswith(os.path.sep):
        member_names.append(member_names[-1] + os.path.sep)
        
    for name in member_names:
        try:
            tar_member = tar.getmember(name)
        except KeyError:
            continue
        break
    else:
        raise AnsibleError("Unable to extract '%s' from collection" % member_names[0])
```

Potentially allows us to try more options in the future if need be.

Reply:
I need to head out for today. I will try to poke at this a bit tomorrow even though it's technically a holiday. If not, I will get to it on Monday.

Reply:
I ended up going with this solution. I'm trying to work out why it's not breaking out of the loop and adjusting the tests.

Reply:
Found my bug. 😊